### PR TITLE
fix: format IPv6 hosts in device URLs

### LINKED ISSUE #1581: bug:  `asynceapi.Device` builds an invalid eAPI URL for IPv6 hosts
### 📦 ANTA version

```console
anta version 1.9.0
```

### 🧰 Installation method

pip

### 🐍 Python version

Python 3.14.6

### 📝 What happened?

`asynceapi.Device.__init__` (`asynceapi/device.py`) builds the eAPI base URL
via an f-string, without wrapping IPv6 host literals in square brackets. For
any host string containing a `:` (i.e. every IPv6 address), the subsequent
`httpx.URL(...)` construction raises `httpx.InvalidURL`, because everything
after the last `:` is interpreted as the port.

## Reproduction (minimal, no real device needed)

```python
import httpx

host = "fd00:c1ab:8::4"
proto = "https"
port = 443
httpx.URL(f"{proto}://{host}:{port}")
```

```
Traceback (most recent call last):
  ...
httpx.InvalidURL: Invalid port: 'c1ab:8::4:443'
```

Reproduction via the public `anta` API (identical error when instantiating an
`AsyncEOSDevice`/`asynceapi.Device` with an IPv6 host):

```python
from asynceapi import Device

Device(host="fd00:c1ab:8::4", username="admin", password="admin")
# httpx.InvalidURL: Invalid port: 'c1ab:8::4:443'
```

In practice, the error occurs on every `anta nrfu` run against an inventory
containing IPv6 hosts, right in the connection/NRFU startup phase, before a
single test is executed.

## Root cause

`asynceapi/device.py`, `Device.__init__` (two locations):

```python
kwargs.setdefault("base_url", httpx.URL(f"{proto}://{self.host}:{self.port}"))
...
login_url = f"{proto}://{self.host}:{self.port}{self.EAPI_LOGIN_URL}"
```

Both f-strings interpolate `self.host` directly into the URL authority. Per
RFC 3986 §3.2.2, an IPv6 address literal in a URL authority **must** be
enclosed in square brackets (`https://[::1]:443`); otherwise the address's
own colons become ambiguous with the port separator — which is exactly why
`httpx.URL` correctly rejects it here.

## Proposed fix

A small helper that brackets IPv6 literals before interpolation (hostnames
and IPv4 literals are returned unchanged):

```python
def _format_host_for_url(host: str) -> str:
    """Return `host` formatted for use in a URL authority component.

    IPv6 literals must be wrapped in square brackets in a URL authority
    (RFC 3986 section 3.2.2), otherwise the trailing ``:<port>`` is
    ambiguous with the address's own colons. Hostnames and IPv4 literals
    are returned unchanged.
    """
    if ":" in host and not host.startswith("["):
        return f"[{host}]"
    return host
```

... used at both locations above (`_url_host = _format_host_for_url(host) if
host else host`, then `f"{proto}://{_url_host}:{self.port}"`). `self.host`
itself is left unchanged (it's used elsewhere only for logging/repr and for
`EapiSessionAuth`, not for URL construction).

Full, verified patch (unified diff, against `anta==1.9.0`,
`asynceapi/device.py`) attached below.

<details>
<summary>Full patch (unified diff)</summary>

```diff
--- a/asynceapi/device.py
+++ b/asynceapi/device.py
@@ -42,6 +42,19 @@
 __all__ = ["Device"]
 
 
+def _format_host_for_url(host: str) -> str:
+    """Return `host` formatted for use in a URL authority component.
+
+    IPv6 literals must be wrapped in square brackets in a URL authority
+    (RFC 3986 section 3.2.2), otherwise the trailing ``:<port>`` is
+    ambiguous with the address's own colons. Hostnames and IPv4 literals
+    are returned unchanged.
+    """
+    if ":" in host and not host.startswith("["):
+        return f"[{host}]"
+    return host
+
+
 # -----------------------------------------------------------------------------
 #
 #                                 CODE BEGINS
@@ -113,7 +126,8 @@
         self.host = host
         self._use_session_auth = use_session_auth
         self._session_auth: EapiSessionAuth | None = None
-        kwargs.setdefault("base_url", httpx.URL(f"{proto}://{self.host}:{self.port}"))
+        _url_host = _format_host_for_url(host) if host else host
+        kwargs.setdefault("base_url", httpx.URL(f"{proto}://{_url_host}:{self.port}"))
         kwargs.setdefault("verify", False)
         if self._use_session_auth:
             if not (username and password):
@@ -122,7 +136,7 @@
             if not self.host:
                 msg = "host is required for session authentication"
                 raise ValueError(msg)
-            login_url = f"{proto}://{self.host}:{self.port}{self.EAPI_LOGIN_URL}"
+            login_url = f"{proto}://{_url_host}:{self.port}{self.EAPI_LOGIN_URL}"
             self._session_auth = EapiSessionAuth(host=self.host, username=username, password=password, login_url=login_url)
             kwargs.setdefault("auth", self._session_auth)
             LOGGER.debug("Device %s: eAPI session-based authentication enabled", self.host)
```

</details>

## Fix verification

With the patch applied:

```python
>>> from asynceapi import Device
>>> d = Device(host="fd00:c1ab:8::4", username="admin", password="admin")
>>> d.base_url
URL('https://[fd00:c1ab:8::4]')
```

- IPv4 and hostname cases are unaffected (`https://192.0.2.1`,
  `https://switch1.example.com` — checked, no regression).
- `anta nrfu -i <ipv6-inventory> -c <catalog> table` then ran successfully
  for the first time against a cEOS-based containerlab topology with
  IPv6-only management (BGP/VXLAN tests passing) — previously the connection
  phase failed with `httpx.InvalidURL` before a single test could run.

## Environment / context

- The fabric under test is deliberately IPv6-only (no IPv4 on the management
  network) — a fully legitimate, RFC-compliant use case, not an edge
  configuration.
- `port_check_url()` (`asynceapi/aio_portcheck.py`) is **not** affected — that
  function operates on an already-constructed `httpx.URL` object, whose
  `.host` property correctly strips brackets itself.
- `EapiSessionAuth` (`asynceapi/_auth.py`) does not build a URL itself; it
  receives a ready-made `login_url` — which is automatically correct once the
  above fix is applied.

### 📋 Relevant output

```shell

```

### PR DESCRIPTION
**Pull request by Sanjays2402**
## Description

IPv6 literals were interpolated into eAPI URLs without brackets, causing `httpx.InvalidURL` before a connection could start. This formats IPv6 hosts for both the base URL and session-login URL while leaving IPv4 addresses and hostnames unchanged.

Fixes #1581

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [ ] I have run pre-commit for code linting and typing (`pre-commit run`)
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes (`tox -e testenv`)

## Summary by CodeRabbit

- **Bug Fixes**
  - Fixed device URL generation when the configured host is an IPv6 address.
  - Session-based authentication now uses properly bracketed IPv6 hosts when constructing login URLs.

- **Tests**
  - Added unit tests covering `base_url` and session login URL formatting for IPv6, including normalization behavior and edge cases (IPv4/hostnames and non-IPv6 colon-containing values).

### CODE REVIEW

File: asynceapi/device.py [hunk 1]
```diff
@@ -113,7 +127,8 @@ def __init__(
         self.host = host
         self._use_session_auth = use_session_auth
         self._session_auth: EapiSessionAuth | None = None
-        kwargs.setdefault("base_url", httpx.URL(f"{proto}://{self.host}:{self.port}"))
+        url_host = _format_url_host(self.host)
+        kwargs.setdefault("base_url", httpx.URL(f"{proto}://{url_host}:{self.port}"))
```
Comment (vitthalmagadum) [inline asynceapi/device.py]: constructs a URL using self.host, even when it is None. Should we guard this like: 
```
if "base_url" not in kwargs:
    if host is None:
        raise ValueError("host is required when base_url is not provided")
    kwargs["base_url"] = ..
```
I know while initializing inventory we have check.

Comment (vitthalmagadum) [reply asynceapi/device.py]:
```
>>> from asynceapi import Device
>>> d = Device(host=None,  username="admin", password="admin")
>>> d.base_url
URL('https://none')
>>> 
```

Comment (gmuloc) [reply asynceapi/device.py]: addressed in [4289700](https://github.com/aristanetworks/anta/pull/1582/commits/4289700f6f8e4c9a0ed715318afc332606803d2f)

**Review by gmuloc — APPROVED** [review]
@Sanjays2402 thanks for creating the PR. I have added a few commits based on review and parsing hardening. Will merge this and it will be part of ANTA 1.10.0

**Review by vitthalmagadum — APPROVED** [review]
